pkg/controllers/provisioning/provisioner.go

```diff
@@ -372,14 +373,185 @@ func (p *Provisioner) getDaemonSetPods(ctx context.Context) ([]*v1.Pod, error) {
+	}
+
+	limitRageMap := map[string][]v1.LimitRange{}
+	for _, lr := range limitRangeList.Items {
+		limitRageMap[lr.Namespace] = append(limitRageMap[lr.Namespace], lr)
+	}
+
 	var ret []*v1.Pod
 	for _, daemonSet := range daemonSetList.Items {
-		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
-		ret = append(ret, pod)
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: daemonSet.Namespace},
+			Spec: *daemonSet.Spec.Template.Spec.DeepCopy()}
+		if applyLR(limitRageMap, pod) == nil {
+			ret = append(ret, pod)
+		}
 	}
 	return ret, nil
 }
 
+// Check and apply limitRange configuration on DaemonSet
+func applyLR(limitRanges map[string][]v1.LimitRange, pod *v1.Pod) error {
+	namespaceedLimitRanges, namespaceExists := limitRanges[pod.ObjectMeta.Namespace]
+	if !namespaceExists {
+		return nil
+	}
+
+	// LimitRanges for Containers
+	for _, lr := range namespaceedLimitRanges {
+		for _, limitrangeItem := range lr.Spec.Limits {
+			if limitrangeItem.Type == v1.LimitTypePod {
+				if err := applyLRTypePod(lr.Name, limitrangeItem, pod); err != nil {
+					return err
+				}
+			} else {
+				if err := applyLRTypeContainer(lr.Name, limitrangeItem, pod); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// Limit Ranges for Pods
+func applyLRTypePod(name string, LRItem v1.LimitRangeItem, pod *v1.Pod) error {
+	podTotalLimits := resources.LimitsForPods(pod)
+	podTotalRequests := resources.RequestsForPods(pod)
+	delete(podTotalLimits, "pods")
+	delete(podTotalRequests, "pods")
+
+	if !resources.Fits(podTotalLimits, LRItem.Max) && LRItem.Max != nil {
+		return fmt.Errorf("the pod limits are not within the Limit Range Maximum (%s)", name)
+	}
+	if !resourceLRGreater(podTotalLimits, LRItem.Min) && LRItem.Min != nil {
+		return fmt.Errorf("the pod limits are not within the Limit Range Minimum (%s)", name)
+	}
+	if !resourceLRRatio(podTotalLimits, podTotalRequests, LRItem.MaxLimitRequestRatio) && LRItem.MaxLimitRequestRatio != nil {
+		return fmt.Errorf("the pod limits are not within the Limit Request Ratio (%s)", name)
+	}
+
+	return nil
+}
+
+// Limit Ranges for Containers
+func applyLRTypeContainer(name string, LRItem v1.LimitRangeItem, pod *v1.Pod) error {
+	if !checkLRMax(LRItem.Max, pod) {
+		return fmt.Errorf("the pod container limits are not within the Limit Range Maximum (%s)", name)
+	}
+	if !checkLRMin(LRItem.Min, pod) {
+		return fmt.Errorf("the pod container limits are not within the Limit Range Minimum (%s)", name)
+	}
+	if !checkLRMaxLimitRequestRatio(LRItem.MaxLimitRequestRatio, pod) {
+		return fmt.Errorf("the pod container limits are not within the Limit Request Ratio (%s)", name)
+	}
+	applyLRDefault(LRItem.Default, pod)
+	applyLRDefaultRequest(LRItem.DefaultRequest, pod)
+
+	return nil
+}
+
+func applyLRDefault(limitrangeDefault v1.ResourceList, pod *v1.Pod) {
+	if limitrangeDefault == nil {
+		return
+	}
+
+	for index := range pod.Spec.Containers {
```

Comment:
This needs to be resource specific, right? E.g. they could set a CPU request/limit, but not a memory one and we only want to default memory in that case.